ingest/expression_files/expression_files.py

```diff
@@ -64,6 +64,41 @@ def set_dataArray(self):
         Each expression file will have its own implementation of setting the
         DataArray with expression data."""
 
+    @staticmethod
+    def has_unique_cells(cell_names: List, study_id, client):
+        """Checks cell names against database to confirm matrix contains unique
+            cell names
+
+         Parameters:
+            cell_names (List[str]): List of cell names in matrix
+            study_id (ObjectId): The study id the cell names belong to
+            client : MongoDB client
+        """
+        COLLECTION_NAME = "data_arrays"
+        query = {
+            "$and": [
+                {"linear_data_type": "Study"},
+                {"array_type": "cells"},
+                {"study_id": study_id},
+            ]
+        }
+        # Returned fields from query results
+        field_names = {"values": 1, "_id": 0}
+        # Dict = {values_1: [<cell names>]... values_n:[<cell names>]}
+        query_results: List[Dict] = list(
+            client[COLLECTION_NAME].find(query, field_names)
+        )
+        # Query did not return results
+        if not query_results:
+            return True
+        # Flatten query results
+        existing_cells = [
+            values
+            for cell_values in query_results
+            for values in cell_values.get("values")
+        ]
```

Comment:
we're going to want the error that we provide to the user to contain at least one of the duplicate cell names, right?  So I suggest this method also return a tuple of boolean/string.  (And only return the first few duplicate names if there are too many to print.  e.g.  "Found X duplicate cells corresponding to already existing expression matrices, including foo1, foo2..."

Reply:
Tracked [here](https://broadworkbench.atlassian.net/browse/SCP-2660)

Reply:
You took care of it :D